packages/shared/src/UI/components/SettingsBoard/hooks/useCallbackOnce.ts

```diff
@@ -0,0 +1,12 @@
+import { DependencyList, useCallback, useRef } from 'react'
+
+export function useCallbackOnce<T extends Function>(callback: T, deps: DependencyList) {
+    const flag = useRef<boolean>(false)
+    const callback_ = (...args: any[]) => {
+        if (flag.current) return
+        flag.current = true
+        callback(...args)
+    }
+
+    return useCallback(callback_ as unknown as T, deps.concat(flag))
```

Comment:
`flag` is a `ref`, no need to concat it.
```suggestion
    return useCallback(callback_ as unknown as T, deps)
```